rook/doc/rook.tex

```diff
@@ -0,0 +1,81 @@
+\documentclass{article}
+
+\title{Rook Test Description}
+
+\begin{document}
+\maketitle
+
+\section{Testers}
+
+The testers are used to run a test.  The tester can fail the test, or
+it can be failed by one of the differs that are attached as subnodes
+in the test input.
+
+\subsection{Common Features}
+
+These are the features that are common to all testers.
+
+\begin{description}
+  \item[type] The type of this test
+  \item[skip] If true skip test
+  \item[prereq] list of tests to run before running this one
+  \item[max\_time] Maximum time that test is allowed to run in seconds
+  \item[os\_max\_time] Maximum time by os. Example: Linux 20 Windows 300 OpenVMS 1000
+  \item[heavy] If true, run only with heavy tests
+  \item[output] Output of the test
+  \item[expected\_fail] if true, then the test should fails, and if it passes, it fails
+  \item[run\_types] The run types that this test is
+  \item[output\_wait\_time] Number of seconds to wait for output
+\end{description}
+
+\subsection{GenericExecutable}
+
+This tester runs an executable, and the test passes if the executable
+returns a 0 as the return code.  This test can also use the differs to
+check for output.
+
+\begin{description}
+  \item[executable] The executable to use
+  \item[parameters] arguments to the executable
+\end{description}
+
+\section{Differs}
+
+\subsection{Common Features}
+
+\begin{description}
+  \item[type] The type of this differ
+  \item[output] Output files to check
+  \item[gold\_files] Gold filenames
+\end{description}
+
+\subsection{Exists}
+
+This checks that a file exists, and if it does, it passes.  There are
+no specific tags for the Exists differ.
+
+\subsection{XML}
+
+This can compare two XML files to see if they are the same.
+
+\begin{description}
+  \item[unordered] if true allow the tags in any order
+  \item[zero\_threshold] it represents the value below which a float is considered zero
+  \item[remove\_whitespace] Removes whitespace before comparing xml node text if True
+  \item[remove\_unicode\_identifier] if true, then remove u infront of a single quote
+  \item[xmlopts] Options for xml checking
```

Comment:
Do we know what acceptable options are? Maybe this is a separate PR to fill this in.

Reply:
Hm, looking they seem to be unordered, rel_err, zero_threshold, remove_whitespace, and remove_unicode_identifier, all of which are already top level options at this point.  I think possibly we should have a separate pull request to remove xmlopts.  I added: Deprecated, the options should be used directly instead of using this (as in use the unordered tag instead of putting it in xmlopts)